tutorials/scripting/gdextension/gdextension_cpp_example.rst

```diff
@@ -0,0 +1,652 @@
+
+.. image:: img/gdextension_cpp_sprite.png
+
+We're finally ready to run the project:
+
+.. image:: img/gdextension_cpp_animated.gif
+
+Adding properties
+-----------------
+
+GDScript allows you to add properties to your script using the ``export``
+keyword. In GDExtension you have to register the properties with a getter and
+sett function or directly implement the ``_get_property_list``, ``_get`` and
+``_set`` methods of an object (but that goes far beyond the scope of this
+tutorial.
+
+Lets add a property that allows us to control the amplitude of our wave.
+
+In our ``gdexample.h`` file we need to add a member variable and getter and setter
+functions:
+
+.. code-block:: C++
+
+    ...
+    private:
+        float time_passed;
+        float amplitude;
+
+    public:
+        void set_amplitude(const float amplitude);
+        float get_amplitude() const;
+    ...
+
+In our ``gdexample.cpp`` file we need to make a number of changes, we will only
+show the methods we end up changing, don't remove the lines we're omitting:
+
+.. code-block:: C++
+
+    void GDExample::_bind_methods() {
+        ClassDB::bind_method(D_METHOD("get_amplitude"), &GDExample::get_amplitude);
+        ClassDB::bind_method(D_METHOD("set_amplitude", "p_amplitude"), &GDExample::set_amplitude);
+
+        ClassDB::add_property("GDExample", PropertyInfo(Variant::FLOAT, "amplitude"), "set_amplitude", "get_amplitude");
+    }
+
+    void GDExample::GDExample() {
+        // initialize any variables here
+        time_passed = 0.0;
+        amplitude = 10.0;
+    }
+
+    void GDExample::_process(float delta) {
+        time_passed += delta;
+
+        Vector2 new_position = Vector2(
+            amplitude + (amplitude * sin(time_passed * 2.0)),
+            amplitude + (amplitude * cos(time_passed * 1.5))
+        );
+
+        set_position(new_position);
+    }
+
+    void GDExample::set_amplitude(const float p_amplitude) {
+        amplitude = p_amplitude;
+    }
+
+    float GDExample::get_amplitude() const {
+        return amplitude;
+    }
+
+Once you compile the module with these changes in place, you will see that a
+property has been added to our interface. You can now change this property and
+when you run your project, you will see that our Godot icon travels along a
+larger figure.
+
+Let's do the same but for the speed of our animation and use a setter and getter
+function. Our ``gdexample.h`` header file again only needs a few more lines of
+code:
+
+.. code-block:: C++
+
+    ...
+        float amplitude;
+        float speed;
+    ...
+        void _process(float delta) override;
+        void set_speed(float p_speed);
+        float get_speed();
+    ...
+
+This requires a few more changes to our ``gdexample.cpp`` file, again we're only
+showing the methods that have changed so don't remove anything we're omitting:
+
+.. code-block:: C++
+
+    void GDExample::_bind_methods() {
+        ClassDB::bind_method(D_METHOD("get_speed"), &GDExample::get_speed);
```

Comment:
might want to add some `...` to indicate we're not removing the code in the fragment above.